src/Overlay.react.js

```diff
@@ -51,14 +52,36 @@ class Overlay extends React.Component {
   }
 
   componentWillReceiveProps(nextProps) {
-    const {onMenuHide, onMenuShow, show} = nextProps;
+    const {
+      a11yStatusContainerId,
+      a11yStatus,
+      multiple,
+      onMenuHide,
+      onMenuShow,
+      resultsCount,
+      selectionCount,
+      show,
+    } = nextProps;
 
     if (this.props.show && !show) {
       onMenuHide();
+      //updateA11yStatus();
     }
 
     if (!this.props.show && show) {
       onMenuShow();
+      updateA11yStatus(true);
+    }
+
+    if (this.props.resultsCount !== nextProps.resultsCount)
+      updateA11yStatus();
+
+    function updateA11yStatus(menuShow = false) {
```

Comment:
Breaking out of the React paradigm here feels like the wrong approach. I'm not sure what the right approach is, but usually when there's a timing issue solved by `setTimeout`, it has something to do with the async nature of `setState`.